rewrite chef checks to use Ridley instead of deprecated chef rest api.

annshah opened the pull request:
## Pull Request Checklist

**Is this in reference to an existing issue?**
rewrite plugin checks to use ridley  issue #25
#### General

- [x] Update Changelog following the conventions laid out on [Keep A Changelog](http://keepachangelog.com/)

- [x] Update README with any necessary configuration snippets

- [x] Binstubs are created if needed

- [x] RuboCop passes

- [ ] Existing tests pass 

#### Purpose

#### Known Compatablity Issues

annshah committed:
changed files for chef library to ridley and added command line option ignoring ssl

annshah committed:
removed comment error

majormoses commented on bin/check-chef-node.rb:
```diff
@@ -65,12 +70,16 @@ def run
   end
 
-  private
+  private 
```
was there a reason for this whitespace?

majormoses commented:
@annshah thanks for the contribution overall this looks good, if you can kill that extra whitespace and fix the gem dependency issues after some quick verification tests I would love to get this merged.

majormoses commented on bin/check-chef-node.rb:
```diff
@@ -50,13 +50,18 @@ class ChefNodeChecker < Sensu::Plugin::Check::CLI
          short: '-K CLIENT-KEY',
          long: '--keys CLIENT-KEY'
 
+  option :ignore_ssl_warnings,
```
I think this would be more appropriately named `ignore_ssl_verification`

majormoses commented on bin/check-chef-nodes.rb:
```diff
@@ -68,19 +68,24 @@ class ChefNodesStatusChecker < Sensu::Plugin::Check::CLI
          long: '--exclude-nodes EXCLUDE-NODES',
          default: '^$'
 
+  option :ignore_ssl_warnings,
```
I think this would be more appropriately named `ignore_ssl_verification`

majormoses commented on sensu-plugins-chef.gemspec:
```diff
@@ -40,7 +40,7 @@ Gem::Specification.new do |s|
-  s.add_runtime_dependency 'ridley',       '4.1.2'
+  s.add_runtime_dependency 'ridley',       '5.1.0'
   s.add_runtime_dependency 'sensu-plugin', '~> 1.2'
   s.add_runtime_dependency 'varia_model', '0.4.1'
```
looks like this needs to be bumped per travis: https://travis-ci.org/sensu-plugins/sensu-plugins-chef/jobs/230068843

annshah committed:
Renamed ignore_ssl_warnings to ignore_ssl_verification, removed whitespace, changed varia_model to 0.6

majormoses commented:
I see the issue, basically ridley 5.x requires buff-extension 2.x which requires >= ruby 2.2.

@eheydrick did sensu ever ship with ruby 2.1? In this case I think we should consider dropping support and do a major bump as this solves issues for lots of users of this gem and with a rewrite I would hate to force using an old version of ridley that we are gonna have to update agains shortly anyways. Thoughts?

annshah commented:
I was facing compatibility issues with earlier versions of ridley. So I believe using latest version might be a necessity.

majormoses commented:
Ya I looked at it and im favor of this breaking change but would like to hear from another maintainer before merging and releasing this one.

majormoses commented:
@cwjohnston thoughts?

cwjohnston commented:
@majormoses per [sensu-build commit history](https://github.com/sensu/sensu-build/commits/master/recipes/ruby.rake) I don't think we've shipped a Sensu release with Ruby 2.1. 

I find these changes agreeable for the most part. I also like the idea of implementing a module library like you mentioned in https://github.com/sensu-plugins/sensu-plugins-chef/issues/25#issuecomment-298467486. Would that be reasonable here to dry things up and make it easier to move to https://github.com/sethvargo/chef-api when that becomes a stable option?

majormoses commented:
@cwjohnston awesome thats what I was looking for! While I would like to force the connection information out right now to me the priority is to get this fixed and nice cleanup later. We should keep the original issue open to track that but I really want this merged asap.

cwjohnston commented:
@majormoses @annshah let's remove ruby 2.1 and add ruby 2.4 to the travis matrix, if those tests pass we can shipit :shipit: 

edit: with a major version bump 👍

majormoses committed:
removed travis testing for ruby 2.1 and added CHANGELOG entry:
- removed ruby 2.1 support
- documented breaking change, impact, and upgrade path that this breaks their setup.
- added ruby 2.4 testing

majormoses commented:
@annshah I see you did not give maintainers access to your fork can you please add this:
```
$ git diff origin/master 
diff --git a/.travis.yml b/.travis.yml
index cf6c5ac..fa96a2a 100644
--- a/.travis.yml
+++ b/.travis.yml

@@ -5,7 +5,6 @@ install:
 - gem install bundler -v 1.12
 - bundle install
 rvm:
-- 2.1
 - 2.2
 - 2.3.0
 notifications:
diff --git a/CHANGELOG.md b/CHANGELOG.md
index d6a3a86..39b5372 100644
--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,8 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 This CHANGELOG follows the format listed at [Keep A Changelog](http://keepachangelog.com/)
 
 ## [Unreleased]
+### Breaking Change
+- re-wrote all checks to use latest ridley. As ridley 5.x requires buff-extension 2.x which requires >= ruby 2.2 and sensu has never shipped with ruby 2.1 the decision was made to drop this support. If you are not using an embedded ruby and are using 2.1 this will break for you. To avoid this you can either switch to embedded ruby or upgrade your ruby to at least 2.2
 
 ## [2.0.1] - 2017-04-28
 ### Fixed
```

Also add 2.4 as requested by @cwjohnston

annshah committed:
removed ruby 2.1 from .travis.yml and updated CHANGELOG.md

annshah committed:
Merge branch 'ridley' of https://github.com/annshah/sensu-plugins-chef into ridley

annshah committed:
add ruby 2.4.1

annshah commented:
@majormoses : I have added all maintainers and Ruby 2.4.1.

majormoses commented:
@annshah thanks I will squash and merge this
